cli/azd/pkg/auth/manager.go

```diff
@@ -429,6 +446,9 @@ func (m *Manager) newCredentialFromFederatedTokenProvider(
 	options := &azidentity.ClientAssertionCredentialOptions{
 		ClientOptions: policy.ClientOptions{
 			Transport: m.httpClient,
+			// TODO: Inject client options instead? this can be done if we're OK
+			// using the default user agent string.
```

Comment:
If you just want to mark requests as originating from azd, you can set `ClientOptions.Telemetry.ApplicationID`. That gets a value like `<your text here> azsdk-go-azidentity/v1.6.0-beta.2 (go1.21.6; linux)`

Reply:
Adding this work item to handle user agent scenarios https://github.com/Azure/azure-dev/issues/2723